backend/service-graphql/src/main/kotlin/dev/johnoreilly/confetti/backend/graphql/DataStoreDataSource.kt

```diff
@@ -144,10 +161,30 @@ class DataStoreDataSource(private val conf: String, private val uid: String? = n
         )
     }
 
-    override fun speakers(): List<Speaker> {
-        return _speakers
+    override fun speakers(first: Int, after: String?): SpeakerConnection {
+        val drop = if (after == null) {
+            0
+        } else {
+            _speakers.indexOfFirst { it.id == after }
+                .also {
+                    if (it == -1) {
+                        return SpeakerConnection(emptyList(), PageInfo(null))
+                    }
+                } + 1
+        }
+        return SpeakerConnection(
+            nodes = _speakers.drop(drop).take(first),
```

Comment:
I always assumed `.take()` would throw if passed an argument longuer than the list. Turns out it's not. TIL!